lib/request.js

```diff
@@ -120,7 +120,9 @@ function buildRequest (opts) {
       done(new Error('unix socket not supported with undici yet'))
       return
     } else {
-      pool = undiciAgent.get(baseUrl || opts.url.origin)
+      const options = getUndiciOptions(opts.undici)
+      options.dispatcher = undiciAgent
```

Comment:
are you sure this is needed?

Reply:
I think so, because of the previous instruction `pool = undiciAgent.get`